src/db.rs

```diff
@@ -710,6 +704,7 @@ impl DbInner {
 		while self.enact_logs(false)? {};
 		self.flush_logs(0)?;
 		while self.process_commits()? {};
+		while self.process_reindex()? {};
```

Comment:
@arkpar this strikes me as needed when writing this, but can also be cause for slow shutdown. So not an innocent change.

Reply:
A single call to `process_reindex` does not complete the whole reindex, but just a single batch. I think it is fine to leave reindexing unfinished and restart on next startup, as it may be slow indeed.

Reply:
Ah, did not notice a loop. Anyway, I think finishing the reindex may indeed be too slow. Causing the user to force-kill if no additional warning is given.